### DISCUSSION TITLE
Which message queue is recommended by your official？

### DISCUSSION THREAD

**Discussion by dushulin** [milvus-io/milvus#28729] [Q&A and General discussion]
we deploy the pulasr, but it have many componets, maybe kafka have simple structure, which is your recommanded?

**Comment by xiaofan-luan** [comment]
both pulsar and kafka should work well with milvus.

For smaller use cases, use standalone should be much easier

**Reply by dushulin** [reply]
We have set up a cluster version, maybe several million 1024 dem embeddings, for online biz, need high throughout, which one is recommended?

**Reply by yhmo** [reply]
5M 1024 dim embeddings, the original data size is 5M * 1024 * 4 bytes = 20GB

"high throughout" ----------- high insert speed or high search QPS?

- If you want to insert 5M entities quickly, you can use the [bulkinsert](https://milvus.io/docs/bulk_insert.md) interface to import numpy files. In fact, 5M is not a huge dataset, batch insert is also ok. You can insert entities batch by batch, each batch inserts 10k~50k entities.

- If you need high search QPS, cluster is prefered. With more query nodes and collection replicas, you can get higher QPS. For example, deploy 4 query nodes, each query node has 25GB memory, and load the collection with 4 replicas: `collection.load(replica_number=4)`.
